src/Polysemy/RandomFu.hs

```diff
@@ -0,0 +1,133 @@
+
+This module also contains the type-class instances to enable "absorbing"
+MonadRandom, ala Polysemy.MTL.  See the tests for MTL or RandomFu for
+examples of that in use.
+-}
+
+module Polysemy.RandomFu
+  (
+    -- * Effect
+    RandomFu (..)
+
+    -- * Actions
+  , sampleRVar
+  , getRandomPrim
+  , sampleDist
+
+    -- * Interpretations
+  , runRandomSource
+  , runRandomIO
+  , runRandomIOPureMT
+
+    -- * Constraint absorber
+  , absorbMonadRandom
+  )
+where
+
+import           Polysemy
+import           Polysemy.MTL
+
+import           Data.IORef                     ( newIORef )
+import qualified Data.Random                   as R
+import qualified Data.Random.Source            as R
+import qualified Data.Random.Internal.Source   as R
+import qualified Data.Random.Source.PureMT     as R
+import           Control.Monad.IO.Class         ( MonadIO(..) )
+
+
+------------------------------------------------------------------------------
+{- | An effect capable of sampling from a "random-fu" RVar or generating a
+single random-variate of any type, @t@ with a
+@Data.Random.Prim t@ constructor, currently one of @Word8@, @Word16@,
+@Word32@, @Word64@, @Double@ or N-byte integer.
+-}
+data RandomFu m r where
+  SampleRVar ::  R.RVar t -> RandomFu m t
+  GetRandomPrim :: R.Prim t -> RandomFu m t
+
+makeSem ''RandomFu
+
+------------------------------------------------------------------------------
+-- | use the 'RandomFu` effect to sample from a "random-fu" @Distribution@.
+sampleDist
+  :: (Member RandomFu r, R.Distribution d t) => d t -> Sem r t
+sampleDist = sampleRVar . R.rvar
+{-# INLINEABLE sampleDist #-}
+
+------------------------------------------------------------------------------
+-- | Run a 'Random' effect using a given 'R.RandomSource'
+runRandomSource
+  :: forall s r a
+   . R.RandomSource (Sem r) s
+  => s
+  -> Sem (RandomFu ': r) a
+  -> Sem r a
+runRandomSource source = interpret $ \case
+    SampleRVar    rv -> R.runRVar (R.sample rv) source
+    GetRandomPrim pt -> R.runRVar (R.getRandomPrim pt) source
+{-# INLINEABLE runRandomSource #-}
+
+------------------------------------------------------------------------------
+-- | Run a 'Random` effect by using the default "random-fu" 'IO' source
+runRandomIO
+  :: forall r a
+   . MonadIO (Sem r)
+  => Sem (RandomFu ': r) a
+  -> Sem r a
+runRandomIO = interpret $ \case
+    SampleRVar    rv -> liftIO $ R.sample rv
+    GetRandomPrim pt -> liftIO $ R.getRandomPrim pt
+{-# INLINEABLE runRandomIO #-}
+
+------------------------------------------------------------------------------
+-- | Run in 'IO', using the given 'R.PureMT' source, stored in an 'IORef'
+runRandomIOPureMT
+  :: MonadIO (Sem r)
+  => R.PureMT
+  -> Sem (RandomFu ': r) a
+  -> Sem r a
+runRandomIOPureMT source re =
+  liftIO (newIORef source) >>= flip runRandomSource re
+{-# INLINEABLE runRandomIOPureMT #-}
+
+------------------------------------------------------------------------------
+absorbMonadRandom
```

Comment:
Haddock for this please; even if it's just to point towards `absorb`.

Reply:
Yep.  Done.